Mathlib/Data/Fintype/Pi.lean

```diff
@@ -163,3 +163,21 @@ theorem Finset.univ_pi_univ {α : Type*} {β : α → Type*} [DecidableEq α] [F
     (Finset.univ.pi fun a : α => (Finset.univ : Finset (β a))) = Finset.univ := by
   ext; simp
 #align finset.univ_pi_univ Finset.univ_pi_univ
+
+/-! ### Diagonal -/
+
+namespace Finset
+variable {ι : Type*} [DecidableEq (ι → α)] {s : Finset α} {f : ι → α}
+
+/-- The diagonal of a finset `s : Finset α` as a finset of functions `ι → α`. -/
```

Comment:
I'm not sure about the placement of this *definition* - it doesn't use fintype anywhere. What do you think about putting the definition in Finset.pi instead? Of course the piFinset lemmas should remain here.

Reply:
Perhaps the choice of filename in #11617 is relevant here?

Reply:
`Data.Finset.Pi` already exists, so I have put the definition there for now